torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -258,11 +305,58 @@ struct GraphFuser {
     at::optional<int> consumer_device = getDevice(consumer);
     Node *real_consumer = consumer->kind() == aten::cat ? consumer->namedInput(attr::tensors)->node() : consumer;
     return isFusable(producer->node()) &&
+      haveSameMapSize(consumer, producer->node()) &&
       allUsersAreThisConsumerOrOccurAfterIt(real_consumer, producer) &&
       consumer_device && consumer_device == getDevice(producer->node()) &&
       (*consumer_device != kCPUDevice || sharedFusionCompiler().canCompileOnCPU());
   }
 
+  value_list tensorInputs(Node * node) {
+    std::vector<Value*> result;
+    result.reserve(node->inputs().size());
+    for (auto * v : node->inputs()) {
+      if (v->type()->isSubtypeOf(DynamicType::get())) {
+        result.push_back(v);
+      }
+    }
+    return result;
+  }
+
+  void maybeInsertExplicitExpands(Node * node) {
+    if (!isSimpleMap(node)) {
+      return;
+    }
+    WithInsertPoint guard(node);
+
+    auto map_size = mapSize(node).value();
+    auto * graph = node->owningGraph();
+
+    auto tensor_inputs = tensorInputs(node);
+    for (auto * producer: tensor_inputs) {
+      auto type = producer->type()->cast<TensorType>();
+      JIT_ASSERT(type);
+      if (std::equal(map_size.begin(), map_size.end(), type->sizes().begin())) {
```

Comment:
You're not checking the length of `type->sizes()` so you can access memory past the end of this array.